### PR TITLE
Support vertex AI (Datasets + Fuse)

### LINKED ISSUE #6491: Support reading from lakeFS in Vertex AI
Vertex can read data either from a mount (using [gcsfuse](https://github.com/GoogleCloudPlatform/gcsfuse)) or using the [Dataset](https://cloud.google.com/python/docs/reference/aiplatform/latest/google.cloud.aiplatform.VideoDataset) API.
lakeFS should support working with both methods, ideally without having to copy data.

### PR DESCRIPTION
**Pull request by ozkatz**
Closes #6491 

This adds a new lua hook function to generate gcsfuse-style symlinks with an example hook showing how to use them to generate symlinks readable from gcsfuse-mounted filesystems.

Additionally, adds docs for the above as well as an example of creating a Vertex Dataset from a lakeFS version.

### CODE REVIEW

**Comment by ozkatz** [comment]
@nopcoder for some more context, here are a few images to illustrate what exporting a branch to gcsfuse would look like:

Commits in a branch:

commit browser

Objects exported in a commit:

object links

(both running in a VM with `gcsfuse --implicit-dirs`.

File: docs/integrations/vertex_ai.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+# Dataset configuration
+lakefs_repo = 'my-repository'
+lakefs_ref = 'main'
+img_dataset = 'datasets/my-images/'
+
+# Vertex configuration
+import_bucket = 'underlying-gcs-bucket'
+
+# produce import file for Vertex's SDK
+files = client.objects.list_objects(lakefs_repo, lakefs_ref, prefix=img_dataset)
```
Comment (nopcoder) [inline docs/integrations/vertex_ai.md:53]: 1. Deprecated the non _api members
```suggestion
files = client.objects_api.list_objects(lakefs_repo, lakefs_ref, prefix=img_dataset)
```
2. Need to handle pagination

File: docs/integrations/vertex_ai.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+rows = []
+for r in files.get('results'):
+    p = PosixPath(r.path)
+    rows.append((r.physical_address, p.parent.name))
+
+# spit out CSV
+csv = '\n'.join([','.join(row) for row in rows])
```
Comment (nopcoder) [inline docs/integrations/vertex_ai.md:60]: import csv and use csv writer to write the csv data.

File: pkg/actions/lua/storage/gcloud/gcs.go [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+}
+
+func (c *GSClient) client() (*storage.Client, error) {
+	var opts []option.ClientOption
+	cred, err := google.CredentialsFromJSON(c.ctx, []byte(c.JSON), googleAuthCloudPlatform)
+	if err != nil {
+		return nil, err
+	}
+	opts = append(opts, option.WithCredentials(cred))
+	return storage.NewClient(c.ctx, opts...)
```
Comment (nopcoder) [inline pkg/actions/lua/storage/gcloud/gcs.go:68]: not using dynamic `opts` - just call `storage.NewClient(c.ctx, option.WithCredentials(cred))`

File: pkg/actions/lua/storage/gcloud/gcs.go [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+		}
+
+		w := obj.NewWriter(c.ctx)
+		w.ObjectAttrs.Metadata = map[string]string{
+			"gcsfuse_symlink_target": physicalAddress,
+		}
+		_, err = w.Write([]byte{})
+		if err != nil {
+			lua.Errorf(l, "could not write object \"%s\": %s", outputAddress, err.Error())
+		}
```
Comment (nopcoder) [inline pkg/actions/lua/storage/gcloud/gcs.go:111]: No need to write nothing.
Close should write the metadata.

**Review by nopcoder — CHANGES_REQUESTED** [review]
LGTM! minor comments

File: docs/integrations/vertex_ai.md [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+
+lakeFS Works with Vertex AI by allowing users to create repositories on [GCS Buckets](../howto/deploy/gcp.md), then use either the Dataset API to create managed Datasets on top of lakeFS version, or by automatically exporting lakeFS versions in a way readable by [Cloud Storage Mounts](https://cloud.google.com/blog/products/ai-machine-learning/cloud-storage-file-system-ai-training). 
+
+{% include toc.html %}
+
+## Using lakeFS with Vertex Managed Datasets
+
+Vertex's [ImageDataset](https://cloud.google.com/python/docs/reference/aiplatform/latest/google.cloud.aiplatform.ImageDataset) and [VideoDataset](https://cloud.google.com/python/docs/reference/aiplatform/latest/google.cloud.aiplatform.VideoDataset) allow creating a dataset by importing a CSV file from gcs (see [`gcs_source`](https://cloud.google.com/python/docs/reference/aiplatform/latest/google.cloud.aiplatform.ImageDataset#google_cloud_aiplatform_ImageDataset_create)).
+
+This CSV file contains gcs addresses of image files and their corresponding labels.
```
[the remark this answers is not in the document]
Comment (ozkatz) [reply docs/integrations/vertex_ai.md]: Fixed

[the remark this answers is not in the document]
Comment (ozkatz) [reply docs/integrations/vertex_ai.md]: Fixed

File: docs/integrations/vertex_ai.md [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+
+Since the lakeFS API supports exporting the underlying gcs address of versioned objects, we can generate such a CSV file when creating the dataset:
```
[the remark this answers is not in the document]
Comment (ozkatz) [reply docs/integrations/vertex_ai.md]: Fixed

File: pkg/actions/lua/storage/gcloud/gcs.go (hunk 3 above)
Comment (ozkatz) [reply pkg/actions/lua/storage/gcloud/gcs.go]: True. Done

File: pkg/actions/lua/storage/gcloud/gcs.go (hunk 4 above)
Comment (ozkatz) [reply pkg/actions/lua/storage/gcloud/gcs.go]: Done. indeed works without it.

**Review by nopcoder — APPROVED** [review]
LGTM! thanks

**Pull request merged**